api_docs/drone_doc.py

```diff
@@ -0,0 +1,212 @@
+"""Generated API Documentation for Drone API using drone_doc_gen.py."""
+
+drone_doc = {
+    "@context": "http://hydrus.com/contexts/droneapi.jsonld",
+    "@id": "http://hydrus.com/droneapi/vocab",
+    "@type": "ApiDocumentation",
+    "description": "API Documentation for the drone side system",
+    "entrypoint": "/droneapi/",
+    "possibleStatus": [],
+    "supportedClass": [
+        {
+            "@context": "http://hydrus.com/context/Drone.jsonld",
+            "@id": "http://hydrus.com/droneapi/vocab/Drone",
+            "@type": "Class",
+            "description": "Class for a drone",
+            "supportedOperation": [
+                {
+                    "@id": ":change_speed",
+                    "@type": "Operation",
+                    "expects": "http://auto.schema.org/speed",
+                    "method": "PUT",
+                    "possibleStatus": [
+                        {
+                            "code": 200,
+                            "description": "Speed Changed"
+                        }
+                    ],
+                    "returns": "null",
+                    "title": "ChangeSpeed"
+                },
+                {
+                    "@id": ":change_position",
+                    "@type": "Operation",
+                    "expects": "http://schema.org/geo",
+                    "method": "PUT",
+                    "possibleStatus": [
+                        {
+                            "code": 200,
+                            "description": "Position Changed"
+                        }
+                    ],
+                    "returns": "null",
+                    "title": "ChangePosition"
+                },
+                {
+                    "@id": ":change_status",
+                    "@type": "Operation",
+                    "expects": "http://schema.org/status",
+                    "method": "PUT",
+                    "possibleStatus": [
+                        {
+                            "code": 200,
+                            "description": "Status Changed"
+                        }
+                    ],
+                    "returns": "null",
+                    "title": "ChangeStatus"
+                },
```

Comment:
I don't think we can support all `change_speed`, `change_position`, `change_status` PUT operations at a single endpoint. We can create different methods for a single type like for speed we can have `change_speed`, `get_speed` etc but I'm not sure how we can use a single endpoint `drone` to handle different things. 
Here's an example of event class from Markus Lanthler's event api 
```json
{
      "@id": "http://schema.org/Event",
      "@type": "hydra:Class",
      "hydra:title": "Event",
      "hydra:description": null,
      "supportedOperation": [
        {
          "@id": "_:event_replace",
          "@type": "http://schema.org/UpdateAction",
          "method": "PUT",
          "label": "Replaces an existing Event entity",
          "description": null,
          "expects": "http://schema.org/Event",
          "returns": "http://schema.org/Event",
          "statusCodes": [
            {
              "code": 404,
              "description": "If the Event entity wasn't found."
            }
          ]
        },
        {
          "@id": "_:event_delete",
          "@type": "http://schema.org/DeleteAction",
          "method": "DELETE",
          "label": "Deletes a Event entity",
          "description": null,
          "expects": null,
          "returns": "http://www.w3.org/2002/07/owl#Nothing",
          "statusCodes": [
          ]
        },
        {
          "@id": "_:event_retrieve",
          "@type": "hydra:Operation",
          "method": "GET",
          "label": "Retrieves a Event entity",
          "description": null,
          "expects": null,
          "returns": "http://schema.org/Event",
          "statusCodes": [
          ]
        }
      ],
      "supportedProperty": [
        {
          "property": "http://schema.org/name",
          "hydra:title": "name",
          "hydra:description": "The event's name",
          "required": true,
          "readonly": false,
          "writeonly": false
        },
        {
          "property": "http://schema.org/description",
          "hydra:title": "description",
          "hydra:description": "Description of the event",
          "required": true,
          "readonly": false,
          "writeonly": false
        },
        {
          "property": "http://schema.org/startDate",
          "hydra:title": "start_date",
          "hydra:description": "The start date and time of the event in ISO 8601 date format",
          "required": true,
          "readonly": false,
          "writeonly": false
        },
        {
          "property": "http://schema.org/endDate",
          "hydra:title": "end_date",
          "hydra:description": "The end date and time of the event in ISO 8601 date format",
          "required": true,
          "readonly": false,
          "writeonly": false
        }
      ]
    }
```

@Mec-iS ^^
One solution is to create a endpoint that receives orders as objects (see [doc](https://docs.google.com/document/d/1BeLoBjT43qhxcUub0MDQ_7OL6yXPVnVWE7-E-CCC-WA/edit))and further processing can be done internally. For example - 
`/droneapi/receive_order` [POST]
```json
{
    "order_id": 123,
    "speed": 30,
    "destination": "xyz"
}